### ISSUE TITLE
CA1834 for nameof expression on single character member

### ISSUE THREAD

**Issue by LWChris** [dotnet/roslyn-analyzers#6499]
### Analyzer

**Diagnostic ID**: [CA1834](https://learn.microsoft.com/de-de/dotnet/fundamentals/code-analysis/quality-rules/ca1834): `Use StringBuilder.Append(char) for single character strings`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET SDK](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: I don't know how I would find this out. Even in the Visual Studio Installer it just says ".NET SDK", so no idea about the version. I'm developing for .NET 6 in Visual Studio 17.4.5 (latest VS 2022), but it has probably been around for longer. 

### Describe the bug

This warning is shown when using a `nameof` expression as argument for the [`StringBuilder.Append`](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append) method and the member in question just so happens to have only one letter (like `X`, `Y` and `Z`, for example). The reason is that `nameof(X)` emits a string and the analyzer thinks I wrote `"X"` myself.

### Steps To Reproduce

```
public readonly record struct Point(double X, double Y, double Z)
{
    public override string ToString()
    {
        var sb = new StringBuilder();
        sb.Append(nameof(Point));
        sb.Append(" { ");
        sb.Append(nameof(X)).Append(" = ").Append(X.ToString("F6")).Append(", ");
        sb.Append(nameof(Y)).Append(" = ").Append(Y.ToString("F6")).Append(", ");
        sb.Append(nameof(Z)).Append(" = ").Append(Z.ToString("F6"));
        sb.Append(" }");
        return sb.ToString();
    }
}
```

### Expected behavior
No warning, since I have no control over the emitted result of a `nameof` expression.

### Actual behavior
The analyzer treats `.Append(nameof(X))` as if I had written `.Append("X")` and emits CA1834 at the location of `nameof`.

## Additional context
There are two conflicting interests here:
1. Use `nameof(X)` over hard-coded strings like `"X"` or `'X'` when referring to a member
2. Use [`StringBuilder.Append(char)`](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append#system-text-stringbuilder-append(system-char)) over [`StringBuilder.Append(string)`](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append#system-text-stringbuilder-append(system-string)) for 1 character string to improve StringBuilder performance.

Unless we can change `nameof` to emit a character in this particular case (not sure if possible, desirable in this case or in general, and if yes where and how to request this), there is no "right" way to handle this to keep the benefits of both `nameof` and `Append(char)`.
